api/internals/init.md

```diff
@@ -0,0 +1,207 @@
+```
+
+## Initialization Sequence
+
+### 1. Verify Initial Conditions
+
+Before initialization can begin, the following must be true:
+
+1. The above input flags are parsed and validated.
+2. The [config file](https://github.com/ipfs/specs/tree/master/repo#config) has not been locked by an active daemon process.
+3. The repo root is writeable.
+4. The repo either doesn't exist and hasn't been initialized, or does already exist but the `--force` flag is supplied.
+
+If all of these are satisfied, output `"initializing ipfs node at %s\n", repoRoot` to standard out and proceed.
+
+### 2. Generate Repo Configuration
+
+These are the steps to producing the default [config tree](https://github.com/ipfs/specs/tree/master/repo#config). The actual output format will depend on the repo backend (i.e. likely JSON if fs-repo). Check the specification for the relevant backend if unsure.
+
+The following sections describe the various subtrees in the config and how to generate them. For simplicity and consistency, JSON will be used.
+
+#### Identity
+
+The RSA keypair must be >= 1024 bits.
+
+Before beginning key generation, write `"generating %v-bit RSA keypair...", nbits` to standard out. (no newline)
+
+Generate the keypair. If successful, append `"done\n"` to standard out.
+
+The private key must be formatted as the base64 encoding of its bytes. It is currently stored unencrypted.
+
+The PeerID must be formatted as
+
+```json
+Base58.Encode(Multihash('SHA2-256', PublicKey.Bytes))
+```
+
+Resulting in the following config tree entry:
+
+```json
+"Identity": {
+	"PeerID": "<PeerID>",
+	"PrivKey": "<PrivKey>"
+}
+```
+
+Finally, write `"peer identity: %s\n", ident.PeerID` to standard out.
+
+#### Bootstrap Peers
+
+The default bootstrap peers form the following config tree:
+
+```json
+"Bootstrap": [
+  // mars.i.ipfs.io
+  "/ip4/104.131.131.82/tcp/4001/ipfs/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ",
+
+  // neptune.i.ipfs.io
+  "/ip4/104.236.176.52/tcp/4001/ipfs/QmSoLnSGccFuZQJzRadHn95W2CrSFmZuTdDWP8HXaHca9z",
+
+  // pluto.i.ipfs.io
+  "/ip4/104.236.179.241/tcp/4001/ipfs/QmSoLPppuBtQSGwKDZT2M73ULpjvfd3aZ6ha4oFGL1KrGM",
+
+  // uranus.i.ipfs.io
+  "/ip4/162.243.248.213/tcp/4001/ipfs/QmSoLueR4xBeUbY9WZ9xGUUxunbKWcrNFTDAadQJmocnWm",
+
+  // saturn.i.ipfs.io
+  "/ip4/128.199.219.111/tcp/4001/ipfs/QmSoLSafTMBsPKadTEgaXctDQVcqN88CNLHXMkTNwMKPnu",
+
+  // venus.i.ipfs.io
+  "/ip4/104.236.76.40/tcp/4001/ipfs/QmSoLV4Bbm51jM9C4gDYZQ9Cy3U6aXMJDAbzgu2fzaDs64",
+
+  // earth.i.ipfs.io
+  "/ip4/178.62.158.247/tcp/4001/ipfs/QmSoLer265NRgSp2LA3dPaeykiS1J6DifTC88f5uVQKNAd",
+
+  // mercury.i.ipfs.io
+  "/ip4/178.62.61.185/tcp/4001/ipfs/QmSoLMeWqB7YGVLJN3pNLQpmmEk35v6wYtsMGLzSr5QBU3",
+
+  // jupiter.i.ipfs.io
+  "/ip4/104.236.151.122/tcp/4001/ipfs/QmSoLju6m7xTh3DuokvT3886QRYqxAzb1kShaanJgW36yx"
+]
+```
+
+#### Supernode Routing
+
+The default supernode routing nodes (for delegated federated routing) form the following config subtree:
+
+```
+SupernodeRouting: {
```

Comment:
I'm not aware of this being currently used of ir it is just a go-ipfs experiment. Might not be necessary to add this to the spec for now (avoiding confusion)